lib/GADS.pm

```diff
@@ -2753,6 +2755,290 @@ prefix '/:layout_name' => sub {
         template 'data' => $params;
     };
 
+    prefix '/report' => sub {
+
+        #view all reports for this instance
+        get '' => require_login sub {
+            my $user   = logged_in_user;
+            my $layout = var('layout') or pass;
+
+            if ( app->has_hook('plugin.linkspace.data_before_request') ) {
```

Comment:
Is this hook needed?

Reply:
I don't know if I'm honest, it's in most other controller requests, and I don't know what it does (if I'm honest), so I left it in where it was and made changes around it! :grimacing: